library/client_events.js

```diff
@@ -137,6 +177,21 @@ module.exports = client => {
           } else {
             updatePracticeRoomChatPermissions(restwo, newMember)
 
+            // run auto-VC creation logic
+            if (areAllPracticeRoomsFull(restwo)) {
+              client.guilds.get(newMember.guild.id).createChannel('Extra Practice Room', 'voice').then(newChan => {
+                // make the new channel go in the right place
+                let categoryChan = client.guilds.get(newMember.guild.id).channels.find(chan => chan.name === 'practice-room-chat').parent
+                newChan.setParent(categoryChan).then(newChan => newChan.setPosition(categoryChan.children.size))
+
+                // gotta update the db
+                restwo['permitted_channels'].push(newChan.id)
```

Comment:
Ah, there's a race condition here. If we get another voiceStatusUpdate before processing finishes we'll create two rooms and only add one of them to the permitted channels list (the second update will squash the first).